synapse/handlers/room_member.py

```diff
@@ -1190,6 +1190,13 @@ async def update_membership_locked(
             origin_server_ts=origin_server_ts,
         )
 
+    async def check_user_membership(self, user_id: str, room_id: str) -> None:
```

Comment:
Name suggestion (maybe this is better :thinking:) `check_for_any_membership_in_room`

Add a docstring

Because both arguments are strings which could be accidentally swapped, we can add `*, ` to force keyword args.

```suggestion
    async def check_for_any_membership_in_room(self, *, user_id: str, room_id: str) -> None:
        """
        Check if the user has any membership in the room and raise error if not.

        Args:
            user_id: The user to check.
            room_id: The room to check.

        Raises:
            AuthError if the user doesn't have any membership in the room.
        """
```